File: Array/OneMatrix.py
from nose.tools import assert_equal

matrix1 = [
	[0, 0, 0, 1],
	[0, 0, 1, 1],
	[0, 1, 1, 1],
	[0, 1, 1, 1]
]

matrix2 = [
	[0, 0, 0, 1],
	[0, 0, 1, 1],
	[1, 1, 1, 1],
	[1, 1, 1, 1]
]

def oneMatrix(matrix):
	count = 0
	i = 0;
	j = len(matrix[0]) - 1
	col = len(matrix[0]) - 1

	while i < len(matrix):
		if matrix[i][j] == 0 or j == 0:
			if matrix[i][j] == 1 and j == 0:
				count += col + 1
			else:
				count += col - j
			i += 1
		elif matrix[i][j] == 1:
			j -= 1

	return count

class OneMatrixTest():
	def test(self, func):
		assert_equal(func(matrix1), 9)
		assert_equal(func(matrix2), 11)
		print('TESTS PASSED')

t = OneMatrixTest()
t.test(oneMatrix)

File: Array/OneMatrix.js
matrix = [
	[0, 0, 0, 1],
	[0, 0, 1, 1],
	[0, 1, 1, 1],
	[1, 1, 1, 1]
];

function matrixOne(matrix){
	var count = 0;
	var i = 0;
	var j = matrix.length - 1;
	var col = matrix.length - 1;
	
	while(i < matrix.length){
		if(matrix[i][j] == 0 || j == 0){
			if(j == 0 && matrix[i][j] == 1){
				count += matrix.length;
			} else{
				count += col - j;
			}
			i++;
		} else{
			j--;
		}
	}

	return count;
}

console.log(matrixOne(matrix));